src/Connection.php

```diff
@@ -618,7 +624,8 @@ public function open()
             $errorNumber,
             $errorDescription,
             $this->connectionTimeout ?: ini_get('default_socket_timeout'),
-            $this->socketClientFlags
+            $this->socketClientFlags,
+            $this->sslContextOptions ? stream_context_create(['ssl' => $this->sslContextOptions]) : stream_context_create()
```

Comment:
Why limit such option to only `ssl`? And shouldn't the default be `null`?

Reply:
Null doesn't work for some reason.  Almost all tests failed with a null.

I'm not sure how relevant rest of the contexts are for this use case. All contexts are listed here: https://www.php.net/manual/en/context.php. For example mongodb or FTP or CURL context is probably not very important in this case. If you think that we should simply expose all of them then let's do that but at least some of them are pretty much useless for this use case.

Reply:
Yes, I can see `stream_context_create()` is used as default. No matter how other options are useful or not it would be really bad if someone needs them and we have to add another piece of code to implement it when we could have it all along. And it would be much more elegant - just `stream_context_create($this->contextOptions)`.

Reply:
Alright. I'll make the change.